### ISSUE TITLE
Links to sections with section number does not work

### ISSUE THREAD

**Issue by pawelkami** [sphinx-contrib/confluencebuilder#346]
I'm using ```sphinxcontrib-confluencebuilder==1.2.0``` from pip.

Confluence builder does not properly publish links to sections of the document when they use section numbers.
It works properly for documents without section numbers.

For example, we have a similar RST document:

```
Some installation manual
----------------------------

This is some installation manual.

Follow these steps to install the software:

#. `Install AAA`_
#. `Configure BBB`_
#. `Install CCC`_


Install AAA
~~~~~~~~~~

aaaaaaaaaaaaaaaaaaaaaa

Configure BBB
~~~~~~~~~~~~~

bbbbbbbbbbbbbbbbbbbbbbbb

Install CCC
~~~~~~~~~~~~

cccccccccccccccccccccc

```

When this is built it looks like that:

```
9. Some installation manual

This is some installation manual.

Follow these steps to install the software:

1. Install AAA (hypertext)
2. Configure BBB (hypertext)
3. Install CCC (hypertext)


9.1. Install AAA

aaaaaaaaaaaaaaaaaaaaaa

9.2. Configure BBB

bbbbbbbbbbbbbbbbbbbbbbbb

9.3. Install CCC

cccccccccccccccccccccc

```

Hypertext links to these subsections look this way:
```http://some.wiki.address/display/namespace/9.+Some+installation+manual#id-9.Someinstallationmanual-InstallAAA```, 
but it doesn't work for confluence, because it expects this link 
```http://some.wiki.address/display/namespace/9.+Some+installation+manual#id-9.Someinstallationmanual-9.1.InstallAAA```.

When I turn off section numbers (```confluence_add_secnumbers = False```), links are generated in the same way, but they work.
Your plugin is missing section number support for links.

**Comment by jdknight** [comment]
The changes outlined in #365 are now available on the main branch and will be available next stable release (v1.3+). Will mark as pending-next-release and will close up when the release has been completed (see #362 for more details).

**Comment by jdknight** [comment]
v1.3 is now available on PyPI -- marking as closed.
